### ISSUE TITLE
InteractTestCase fail

### ISSUE THREAD

**Issue by jlec** [pexpect/pexpect#282]
```python
========================================= test session starts ==========================================
platform linux -- Python 3.4.3, pytest-2.8.1, py-1.4.30, pluggy-0.3.1
rootdir: /var/tmp/portage/dev-python/pexpect-4.0.1/work/pexpect-4.0.1, inifile: setup.cfg
collected 3 items

tests/test_interact.py FFF

=============================================== FAILURES ===============================================
________________________________ InteractTestCase.test_interact_escape _________________________________

self = <tests.test_interact.InteractTestCase testMethod=test_interact_escape>

    def test_interact_escape(self):
        " Ensure `escape_character' value exits interactive mode. "
        p = pexpect.spawn(self.interact_py, timeout=5, env=self.env)
>       p.expect('READY')

tests/test_interact.py:49:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
pexpect/spawnbase.py:315: in expect
    timeout, searchwindowsize, async)
pexpect/spawnbase.py:339: in expect_list
    return exp.expect_loop(timeout)
pexpect/expect.py:102: in expect_loop
    return self.eof(e)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <pexpect.expect.Expecter object at 0x7f8ef11dd6a0>
err = EOF('End Of File (EOF). Exception style platform.',)

    def eof(self, err=None):
        spawn = self.spawn
        from . import EOF

        spawn.before = spawn.buffer
        spawn.buffer = spawn.string_type()
        spawn.after = EOF
        index = self.searcher.eof_index
        if index >= 0:
            spawn.match = EOF
            spawn.match_index = index
            return index
        else:
            spawn.match = None
            spawn.match_index = None
            msg = str(spawn)
            if err is not None:
                msg = str(err) + '\n' + msg
>           raise EOF(msg)
E           pexpect.exceptions.EOF: End Of File (EOF). Exception style platform.
E           <pexpect.pty_spawn.spawn object at 0x7f8ef124b8d0>
E           command: /usr/bin/python3.4
E           args: ['/usr/bin/python3.4', 'interact.py']
E           searcher: None
E           buffer (last 100 chars): b''
E           before (last 100 chars): b"t read '/var/tmp/portage/dev-python/pexpect-4.0.1/work/pexpect-4.0.1/.coveragerc' as a config file\r\n"
E           after: <class 'pexpect.exceptions.EOF'>
E           match: None
E           match_index: None
E           exitstatus: None
E           flag_eof: True
E           pid: 23374
E           child_fd: 12
E           closed: False
E           timeout: 5
E           delimiter: <class 'pexpect.exceptions.EOF'>
E           logfile: None
E           logfile_read: None
E           logfile_send: None
E           maxread: 2000
E           ignorecase: False
E           searchwindowsize: None
E           delaybeforesend: 0.05
E           delayafterclose: 0.1
E           delayafterterminate: 0.1

pexpect/expect.py:49: EOF
----------------------------------------- Captured stdout call -----------------------------------------

 tests.test_interact.InteractTestCase.test_interact_escape
______________________________ InteractTestCase.test_interact_escape_None ______________________________

self = <tests.test_interact.InteractTestCase testMethod=test_interact_escape_None>

    def test_interact_escape_None(self):
        " Return only after Termination when `escape_character=None'. "
        p = pexpect.spawn('{self.interact_py} --no-escape'.format(self=self),
                          timeout=5, env=self.env)
>       p.expect('READY')

tests/test_interact.py:61:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
pexpect/spawnbase.py:315: in expect
    timeout, searchwindowsize, async)
pexpect/spawnbase.py:339: in expect_list
    return exp.expect_loop(timeout)
pexpect/expect.py:102: in expect_loop
    return self.eof(e)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <pexpect.expect.Expecter object at 0x7f8ef1205550>
err = EOF('End Of File (EOF). Exception style platform.',)

    def eof(self, err=None):
        spawn = self.spawn
        from . import EOF

        spawn.before = spawn.buffer
        spawn.buffer = spawn.string_type()
        spawn.after = EOF
        index = self.searcher.eof_index
        if index >= 0:
            spawn.match = EOF
            spawn.match_index = index
            return index
        else:
            spawn.match = None
            spawn.match_index = None
            msg = str(spawn)
            if err is not None:
                msg = str(err) + '\n' + msg
>           raise EOF(msg)
E           pexpect.exceptions.EOF: End Of File (EOF). Exception style platform.
E           <pexpect.pty_spawn.spawn object at 0x7f8ef11ddc88>
E           command: /usr/bin/python3.4
E           args: ['/usr/bin/python3.4', 'interact.py', '--no-escape']
E           searcher: None
E           buffer (last 100 chars): b''
E           before (last 100 chars): b"t read '/var/tmp/portage/dev-python/pexpect-4.0.1/work/pexpect-4.0.1/.coveragerc' as a config file\r\n"
E           after: <class 'pexpect.exceptions.EOF'>
E           match: None
E           match_index: None
E           exitstatus: None
E           flag_eof: True
E           pid: 23375
E           child_fd: 13
E           closed: False
E           timeout: 5
E           delimiter: <class 'pexpect.exceptions.EOF'>
E           logfile: None
E           logfile_read: None
E           logfile_send: None
E           maxread: 2000
E           ignorecase: False
E           searchwindowsize: None
E           delaybeforesend: 0.05
E           delayafterclose: 0.1
E           delayafterterminate: 0.1

pexpect/expect.py:49: EOF
----------------------------------------- Captured stdout call -----------------------------------------

 tests.test_interact.InteractTestCase.test_interact_escape_None
_____________________________ InteractTestCase.test_interact_exit_unicode ______________________________

self = <tests.test_interact.InteractTestCase testMethod=test_interact_exit_unicode>

    def test_interact_exit_unicode(self):
        " Ensure subprocess receives utf8. "
        p = pexpect.spawnu('{self.interact_py} --utf8'.format(self=self),
                           timeout=5, env=self.env)
>       p.expect('READY')

tests/test_interact.py:79:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
pexpect/spawnbase.py:315: in expect
    timeout, searchwindowsize, async)
pexpect/spawnbase.py:339: in expect_list
    return exp.expect_loop(timeout)
pexpect/expect.py:102: in expect_loop
    return self.eof(e)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <pexpect.expect.Expecter object at 0x7f8ef11f4390>
err = EOF('End Of File (EOF). Exception style platform.',)

    def eof(self, err=None):
        spawn = self.spawn
        from . import EOF

        spawn.before = spawn.buffer
        spawn.buffer = spawn.string_type()
        spawn.after = EOF
        index = self.searcher.eof_index
        if index >= 0:
            spawn.match = EOF
            spawn.match_index = index
            return index
        else:
            spawn.match = None
            spawn.match_index = None
            msg = str(spawn)
            if err is not None:
                msg = str(err) + '\n' + msg
>           raise EOF(msg)
E           pexpect.exceptions.EOF: End Of File (EOF). Exception style platform.
E           <pexpect.pty_spawn.spawn object at 0x7f8ef08f1438>
E           command: /usr/bin/python3.4
E           args: ['/usr/bin/python3.4', 'interact.py', '--utf8']
E           searcher: None
E           buffer (last 100 chars): ''
E           before (last 100 chars): "t read '/var/tmp/portage/dev-python/pexpect-4.0.1/work/pexpect-4.0.1/.coveragerc' as a config file\r\n"
E           after: <class 'pexpect.exceptions.EOF'>
E           match: None
E           match_index: None
E           exitstatus: None
E           flag_eof: True
E           pid: 23376
E           child_fd: 14
E           closed: False
E           timeout: 5
E           delimiter: <class 'pexpect.exceptions.EOF'>
E           logfile: None
E           logfile_read: None
E           logfile_send: None
E           maxread: 2000
E           ignorecase: False
E           searchwindowsize: None
E           delaybeforesend: 0.05
E           delayafterclose: 0.1
E           delayafterterminate: 0.1

pexpect/expect.py:49: EOF
----------------------------------------- Captured stdout call -----------------------------------------

 tests.test_interact.InteractTestCase.test_interact_exit_unicode
======================================= 3 failed in 0.74 seconds =======================================
```

for pexpect-3.3 and 4.0.1 for python-2.7/3.3/3.4 and pypy.

**Comment by jquast** [comment]
Thank you @jlec -- Is this gentoo Linux (I see phrase "portage" in output)? We test on debian and ubuntu but have not seen this issue before, do you have any inclination on why the execution of:

``/usr/bin/python3.4 interact.py --utf8`` would fail to output the phrase ``READY`` within 5 seconds?

**Comment by jquast** [comment]
I think I know the error @jlec, but I don't understand it, I believe the full error should read:

``could not read '/var/tmp/portage/dev-python/pexpect-4.0.1/work/pexpect-4.0.1/.coveragerc' as a config file\r\n"``

which is from coverage.py, raising an exception, preventing interact.py from executing further, https://bitbucket.org/ned/coveragepy/src/088704cc33ecaad644a24116477cd8d112d691da/coverage/control.py?at=default&fileviewer=file-view-default#control.py-140

If that helps any, do you know why this might happen in your build environment? I can disable coverage to prevent it from reading .coveragerc in the subprocess, but I'm not completely satisfied to the cause of this issue, as .coveragerc is formatted perfectly fine, checked into vcs at that version with correct file permissions

**Comment by jlec** [comment]
Although this is the portage dir I did the test manually to confirm that it isn't the package manager creating the problem.

```/usr/bin/python3.4 interact.py --utf8```outputs ```READY```

**Comment by jlec** [comment]
regarding the coveragerc I have no idea. I have coverage-4.0 installed.

**Comment by jlec** [comment]
```touch .coveragerc``` works around the problem.

**Comment by jquast** [comment]
.coveragerc exists in VCS at that location, why is it missing in your build process? Are .files excluded from extraction? https://github.com/pexpect/pexpect/blob/4.0.1/.coveragerc

**Comment by jlec** [comment]
I am using the pypi tarball. Currently I am packaging the version bump for Gentoo, not doing development.

**Comment by jquast** [comment]
Ah-ha, you're testing from the pypi tarball. I never expected such, but we can accommodate :)

**Comment by takluyver** [comment]
Closed by #282, though we'll need to make a new release for the PyPI tarball to work for this.
